barretenberg/cpp/src/barretenberg/boomerang_value_detection/graph_description_sha256.test.cpp

```diff
@@ -14,39 +14,65 @@
 
 using namespace bb;
 using namespace bb::stdlib;
+using namespace cdg;
 
 using Builder = UltraCircuitBuilder;
+using byte_array_pt = byte_array<Builder>;
+using packed_byte_array_pt = packed_byte_array<Builder>;
+using field_pt = field_t<Builder>;
 
-using byte_array_ct = byte_array<Builder>;
-using packed_byte_array_ct = packed_byte_array<Builder>;
-using field_ct = field_t<Builder>;
+/**
+ static analyzer usually prints input and output variables as variables in one gate. In tests these variables
+ are not dangerous and usually we can filter them by adding gate for fixing witness. Then these variables will be
+ in 2 gates, and static analyzer won't print them. functions fix_vector and fix_byte_array do it
+ for vector of variables and packed_byte_array respectively
+*/
+
+void fix_vector(std::vector<field_pt>& vector)
+{
+    for (auto& elem : vector) {
+        elem.fix_witness();
+    }
+}
+
+void fix_byte_array(packed_byte_array_pt& input)
+{
+    std::vector<field_pt> limbs = input.get_limbs();
+    fix_vector(limbs);
+}
 
 /**
- all these tests check graph description for sha256 circuits. All circuits have to consist from 1 connected component
+ all these tests check graph description for sha256 circuits. All circuits have to consist of 1 connected component
  */
 
 TEST(boomerang_stdlib_sha256, test_graph_for_sha256_55_bytes)
 {
     // 55 bytes is the largest number of bytes that can be hashed in a single block,
     // accounting for the single padding bit, and the 64 size bits required by the SHA-256 standard.
     auto builder = Builder();
-    packed_byte_array_ct input(&builder, "An 8 character password? Snow White and the 7 Dwarves..");
+    packed_byte_array_pt input(&builder, "An 8 character password? Snow White and the 7 Dwarves..");
+    fix_byte_array(input);
 
-    packed_byte_array_ct output_bits = stdlib::sha256(input);
+    packed_byte_array_pt output_bits = stdlib::sha256(input);
 
-    std::vector<field_ct> output = output_bits.to_unverified_byte_slices(4);
+    std::vector<field_pt> output = output_bits.to_unverified_byte_slices(4);
+    fix_vector(output);
 
     Graph graph = Graph(builder);
     auto connected_components = graph.find_connected_components();
     EXPECT_EQ(connected_components.size(), 1);
+    auto variables_in_one_gate = graph.show_variables_in_one_gate(builder);
+    EXPECT_EQ(variables_in_one_gate.size(), 0);
+    if (variables_in_one_gate.size() > 0) {
+        for (const auto& elem : variables_in_one_gate) {
+            info("elem == ", elem);
```

Comment:
It's better to remove info from tests in the general case, but here it's not an issue since the branch is not being taken